### PR TITLE
[Improve][build] Improved scope of maven-shade-plugin

### LINKED ISSUE #2599: [Improve][build] maven-shade-plugin using scope is too large
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Now I find `maven-shade-plugin` in the root directory of `pom.xml` was imported, causes the entire seatunnel project are shaded, which makes seatunnel after packaging size is very large.

image
image

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
Improved scope of maven-shade-plugin

This is #2599 task
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ashulin** [comment]
ZonedDateTime should be solved by another PR as an improvement.

**Comment by ashulin** [comment]
@CalvinKirs @TyrantLucifer @laglangyue PTAL

**Review by TyrantLucifer — DISMISSED** [review]
LGTM.

**Review by CalvinKirs — COMMENTED** [review]
How about declaring in parent dependency? This way you don't have to declare every submodule

**Comment by EricJoy2048** [comment]
> How about declaring in parent dependency? This way you don't have to declare every submodule

Yes, I think so, too.
Add `maven-shade-plugin` to `seatunnel-connector-v2/pom.xml` instead `seatunnel-connector-v2/connector-xxx/pom.xml` is better.

**Comment by hailin0** [comment]
Need to exclude slf4j, logback, log4j, other... in connector shade

**Comment by ashulin** [comment]
@CalvinKirs @EricJoy2048 If declared in the parent pom, it will cause `connector-common`, `connector-file-base`, etc. to also execute the shaded.

**Comment by EricJoy2048** [comment]
> @CalvinKirs @EricJoy2048 If declared in the parent pom, it will cause `connector-common`, `connector-file-base`, etc. to also execute the shaded.

Can we use `exclude` to exclude `connector-common`, `connector-file-base`?

**Comment by CalvinKirs** [comment]
> > @CalvinKirs @EricJoy2048 If declared in the parent pom, it will cause `connector-common`, `connector-file-base`, etc. to also execute the shaded.
> 
> Can we use `exclude` to exclude `connector-common`, `connector-file-base`?

We can manually declare the plugin in these special modules and then add the`skip` parameter

**Comment by ashulin** [comment]
@CalvinKirs @EricJoy2048  Thanks, I got it. maven-shade-plugin provided the `skip` parameter in the latest version(3.3.0).

**Review by CalvinKirs — APPROVED** [review]

File: seatunnel-connectors-v2/connector-dingtalk/pom.xml [hunk 1]
```diff
@@ -41,6 +41,12 @@
       <groupId>com.aliyun</groupId>
       <artifactId>alibaba-dingtalk-service-sdk</artifactId>
       <version>${dingtalk.service.version}</version>
+      <exclusions>
+        <exclusion>
+          <artifactId>log4j</artifactId>
+          <groupId>log4j</groupId>
+        </exclusion>
+      </exclusions>
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-dingtalk/pom.xml:49]: Excluded due to an error resolving JMX/JMS dependencies at build phase.

**Pull request merged**